src/object.d

```diff
@@ -287,6 +287,24 @@ void setSameMutex(shared Object ownee, shared Object owner)
     _d_setSameMutex(ownee, owner);
 }
 
+/**
+ * Gets default initializer for type T as an untyped array of data.
+ * Params:
+ *      T = type to get initializer for
+ * Returns:
+ *      If the type should be initialized to all
+ *      zeros, an array with a null ptr and a length equal to the type size will
+ *      be returned.
+ *      For static arrays, this returns the default initializer for
+ *      a single element of the array.
```

Comment:
I think this static-array special case is unintuitive and error-prone. I think it's better to disallow static array types via `if` template clause.